fendermint/vm/interpreter/src/chain.rs

```diff
@@ -266,7 +266,11 @@ where
 
                     // error happens if we cannot get the validator set from ipc agent after retries
                     let validator_changes = provider
-                        .validator_changes_from(prev_height + 1, finality.height)
+                        .validator_changes_from(
+                            prev_height + 1,
+                            finality.height,
```

Comment:
* We may need to update the cache not to clean `finality.height - 1` (even if its null we know that the state is final up till here). 
* We need to apply this same approach for top-down messages.

If we receive 100 and we don't have yet that info because we are not up to date, we should either crash or not make progress, but never execute to avoid consensus failures.